server/server.go

```diff
@@ -72,31 +115,60 @@ func (s *Server) loadSandbox(id string) error {
 	})
 	sandboxPath := filepath.Join(s.sandboxDir, id)
 
-	scontainer, err := oci.NewContainer(m.Annotations["ocid/container_id"], m.Annotations["ocid/container_name"], sandboxPath, sandboxPath, labels, id, false)
+	if err := label.ReserveLabel(processLabel); err != nil {
+		return err
+	}
+	if err := label.ReserveLabel(mountLabel); err != nil {
+		return err
+	}
+
+	cname, err := s.reserveContainerName(m.Annotations["ocid/container_id"], m.Annotations["ocid/container_name"])
+	if err != nil {
+		return err
+	}
+	scontainer, err := oci.NewContainer(m.Annotations["ocid/container_id"], cname, sandboxPath, sandboxPath, labels, id, false)
 	if err != nil {
 		return err
 	}
 	s.addContainer(scontainer)
 	if err = s.runtime.UpdateStatus(scontainer); err != nil {
-		logrus.Warnf("error updating status for container %s: %v", scontainer, err)
+		logrus.Warnf("error updating status for container %s: %v", scontainer.ID(), err)
+	}
+	if err = s.ctrIDIndex.Add(scontainer.ID()); err != nil {
+		return err
 	}
 	if err = s.podIDIndex.Add(id); err != nil {
 		return err
 	}
 	return nil
 }
 
-func (s *Server) restore() error {
-	dir, err := ioutil.ReadDir(s.sandboxDir)
+func (s *Server) restore() {
+	sandboxDir, err := ioutil.ReadDir(s.sandboxDir)
 	if err != nil {
-		return err
+		logrus.Warnf("could not read sandbox directory %s: %v", sandboxDir, err)
```

Comment:
should it? if we fail to load a pod/container should we entirely bail out? docker is just skipping that container's load, shall we fail instead?